### PR TITLE
windows feature WS_EX_NOREDIRECTIONBITMAP

### PR DESCRIPTION
**Pull request by ekicyou**
- [x] Tested on all platforms changed
- [x] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [ ] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [ ] Created an example program if it would help users understand this functionality

Windows additionally has `WindowBuilderExt::with_no_redirection_bitmap`.
In order to use Direct Composition API, implemented the function to set WS_EX_NOREDIRECTIONBITMAP extended attribute.

ref: [MSDN Windows with C++ - High-Performance Window Layering Using the Windows Composition Engine](https://msdn.microsoft.com/magazine/dn745861.aspx)

### CODE REVIEW

File: src/os/windows.rs [hunk 1]
```diff
@@ -55,6 +55,9 @@ pub trait WindowBuilderExt {
 
     /// This sets `ICON_BIG`. A good ceiling here is 256x256.
     fn with_taskbar_icon(self, taskbar_icon: Option<Icon>) -> WindowBuilder;
+
+    /// This sets `WS_EX_NOREDIRECTIONBITMAP`.
+    fn with_no_redirection_bitmap(self) -> WindowBuilder;
```
Comment (francesca64) [inline src/os/windows.rs]: It's conventional for builder methods to take a `bool`, for more flexibility.

**Comment by francesca64** [comment]
Well, I don't really know how to test this, since it just makes my OpenGL app transparent.

My only question is if this is mutually exclusive with `with_transparency`. Having both made the client area black.

**Comment by ekicyou** [comment]
> My only question is if this is mutually exclusive with with_transparency. Having both made the client area black.

Excuse my translation.

It becomes exclusive designation.

Both can be used as a means to realize a transparent window, but the realization method is different.

WS_EX_NOREDIRECTIONBITMAP specifies not to display the GDI bitmap.
The GDI bitmap is the old rendering API, not GPU optimized.
By using only composition surfaces directly managed by DWM for screen display, you can intercept maximum GPU optimization of Windows.

As a composition surface, you can specify a swap chain.

I am sorry, I can not prepare a small sample now.
The trial at hand is in trial with the [modification](https://github.com/ekicyou/webrender) of [[servo/webrender]](https://github.com/servo/webrender).

**Comment by ekicyou** [comment]
Thank you. Check the source and fix it.

**Comment by francesca64** [comment]
Alright, looks great.

**Pull request merged**